lib/types/application/ApplicationFormatter.js

```diff
@@ -19,12 +26,33 @@ class ApplicationFormatter extends AbstractFormatter {
 				"/": project.resources.configuration.paths.webapp
 			};
 
-			return this.readManifest(project).then(function(manifest) {
-				if (!manifest["sap.app"] || !manifest["sap.app"].id) {
+			return this.readManifest(project).then((manifest) => {
+				// check for a proper sap.app/id in manifest.json to determine namespace
+				const appId = manifest["sap.app"] && manifest["sap.app"].id;
+				if (!appId) {
 					log.warn(`No "sap.app" ID configuration found in manifest of project ${project.metadata.name}`);
 					return;
 				}
-				project.metadata.namespace = manifest["sap.app"].id.replace(/\./g, "/");
+				// check app id for placeholder and try to read the placeholder from the pom.xml
+				const parts = appId.match(/^\$\{(.*)\}$/);
+				if (parts) {
+					log.warn(`No valid "sap.app" ID found in manifest of project ${project.metadata.name}. Reading "${parts[1]}" from pom.xml...`);
+					return this.readPOM(project).then((pom) => {
+						if (pom.project && pom.project.properties && pom.project.properties[parts[1]]) {
+							return pom.project.properties[parts[1]];
+						} else {
+							let obj = pom;
+							parts[1].split(".").forEach((part) => {
+								obj = obj[part];
+								return !obj;
+							});
+							return obj;
+						}
+					});
+				}
+				return appId;
+			}).then((appId) => {
+				project.metadata.namespace = appId ? appId.replace(/\./g, "/") : undefined;
 			}).catch((err) => {
 				log.verbose(`No manifest found for project ${project.metadata.name}.`);
```

Comment:
This catch needs to move to only catch errors caused by manifest read errors. Currently also catches errors from reading the POM

Reply:
In addition, a similar catch for the POM read is required. The reason for them is to suppress the errors since it is still valid for applications to not have a namespace.

Reply:
Good point... Fixed!